src/vl-mapactions-delete-action.js

```diff
@@ -30,20 +30,24 @@ export class VlDeleteAction extends VlBoxSelectAction {
         layer.getSource().removeFeature(feature);
       }
     };
+    
+    const afterOnDelete = () => {
+    	this.clearFeatures();
+    	this.map.render();
+    };
 
     super(layer, (features) => {
       if (onDelete && onDelete != null) {
         onDelete(features, (feature) => {
           removeFeature(feature);
-          this.clearFeatures();
         }, () => {
-          this.clearFeatures();
         });
+        afterOnDelete();
```

Comment:
nu zit dit wel niet meer in de callback precies? Gaat ge dan niet soms te snel zijn met re-renderen?

Reply:
Nee net later, hij ging nu voor elke feature rerenderen

Reply:
Maar de removeFeature zit in een onDelete callback, en onDelete zou async kunnen dingen doen waardoor uw afterOnDelete toch ook nog steeds voor de callbacks in onDelete wordt opgeroepen?

Reply:
Maar dat "probleem", dat er dingen async zouden kunnen gebeuren in die callback zat er dan ook al in hoor ... Dat heb ik niet gewijzgd.

Reply:
Ik zeg niet per se dat het in een van de callbacks van onDelete moet zitten, maar misschien moet er ge-await worden op onDelete en moet onDelete een promise teruggeven, of iets in die aard.

Reply:
Ah nee, ik ben mee, de client kiest wanneer hij die callback aanroept ...

Reply:
> Maar dat "probleem", dat er dingen async zouden kunnen gebeuren in die callback zat er dan ook al in hoor ... Dat heb ik niet gewijzgd.

Nee, maar dat lijkt me ook een bug, eentje dat voor dezelfde symptomen zou kunnen zorgen, hier?

Reply:
Ja ca va, dan gaat hij wel ne render doen voor elke feature, maar misschien is dat geen groot probleem.

Reply:
Dus: besloten dat het ok is om for now de re-render bij elke delete te doen.
In de toekomst is een mogelijke verbetering om voor elke te verwijderen feature een promise bij te houden, deze te resolven als de feature verwijderd werd en op het einde een await te doen op al de promises vooraleer je afterOnDelete oproept.